src/main/java/spoon/support/reflect/declaration/CtRecordComponentImpl.java

```diff
@@ -36,18 +39,45 @@ public class CtRecordComponentImpl extends CtNamedElementImpl implements CtRecor
 	public CtMethod<?> toMethod() {
 		CtMethod<?> method = this.getFactory().createMethod();
 		method.setSimpleName(getSimpleName());
-		method.setType((CtTypeReference) getType());
+		method.setType(getType() != null ? getType().clone() : null);
```

Comment:
might make sense to have a `cloneType` method considering how often this code is used here

Reply:
That is something that is generally useful, but the getters require returning the original reference